sdk/core/azure_core/README.md

```diff
@@ -1,20 +1,283 @@
+
+Azure SDK client libraries typically expose one or more _service client_ types that
+are the main starting points for calling corresponding Azure services.
+You can easily find these client types as their names end with the word _Client_.
+For example, `SecretClient` can be used to call the Key Vault service and interact with secrets,
+and `KeyClient` can be used to access Key Vault service cryptographic keys.
+
+These client types can be instantiated by calling a simple `new` or `builder` function that takes various configuration options.These options are passed as a parameter that extends `ClientOptions` class exposed by `azure_core`.
+Various service specific options are usually added to its subclasses, but a set of SDK-wide options are
+available directly on `ClientOptions`.
+
+```rust no_run
+use azure_core::ClientOptions;
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{SecretClient, SecretClientOptions};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let options = SecretClientOptions {
+        api_version: "7.0".to_string(),
+        client_options: ClientOptions::default(),
+    };
+
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential.clone(),
+        Some(options),
+    )?;
+
+    Ok(())
+}
+```
+
+### Accessing HTTP Response Details Using `Response<T>`
+
+_Service clients_ have methods that can be used to call Azure services. We refer to these client methods as _service methods_.
+_Service methods_ return a shared `azure_core` type `Response<T>` (in rare cases its non-generic sibling, a raw `Response`).
+This type provides access to both the deserialized result of the service call, and to the details of the HTTP response returned from the server.
+
+```rust no_run
+use azure_core::Response;
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{models::SecretBundle, SecretClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // create a client
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential.clone(),
+        None,
+    )?;
+
+    // call a service method, which returns Response<T>
+    let response: Response<SecretBundle> = client.get_secret("SecretName", "", None).await?;
+
+    // Response<T> has two main accessors.
+    // The into_body() function for accessing the deserialized result of the call
+    let secret = response.into_body().await?;
+
+        // get response again because it was moved in above statement
+    let response: Response<SecretBundle> = client.get_secret("SecretName", "", None).await?;
+
+    // .. and the deconstruct() method for accessing all the details of the HTTP response
+    let (status, headers, body) = response.deconstruct();
```

Comment:
This won't work. `into_body()` takes ownership of `response` i.e., it takes a `self` instead of `&self`. So does `deconstruct`. It's one or the other. Most people won't need/use `deconstruct`.

Reply:
I think this section in the .NET README wanted to show how to get the HTTP response details. Is there a better way than using deconstruct.

Reply:
@heaths, won't redefining the response by fetching it again fix that?